crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -37,68 +51,101 @@ impl CoinGecko {
         api_key: Option<String>,
         chain_id: u64,
     ) -> Result<Self> {
-        let chain = match chain_id {
-            1 => "ethereum".to_string(),
-            100 => "xdai".to_string(),
-            42161 => "arbitrum-one".to_string(),
+        let (chain, native_price) = match chain_id {
```

Comment:
If you inject an instance of WETH9 you will be able to call `weth.symbol` and if it's WETH you can skip native price estimation

Reply:
@fleupold please could you take a look at it? :pray: